Source/Core/DolphinQt/Config/ConfigControls/ConfigBool.cpp

```diff
@@ -5,29 +5,75 @@
 
 #include <QEvent>
 #include <QFont>
+#include <QMouseEvent>
 #include <QSignalBlocker>
 
 #include "Common/Config/Config.h"
 
 #include "DolphinQt/Settings.h"
 
 ConfigBool::ConfigBool(const QString& label, const Config::Info<bool>& setting, bool reverse)
-    : ToolTipCheckBox(label), m_setting(setting), m_reverse(reverse)
+    : ConfigBool(label, setting, nullptr, reverse)
 {
-  connect(this, &QCheckBox::toggled, this, &ConfigBool::Update);
-  setChecked(Config::Get(m_setting) ^ reverse);
+}
+
+ConfigBool::ConfigBool(const QString& label, const Config::Info<bool>& setting,
+                       Config::Layer* target_layer, bool reverse)
+    : ToolTipCheckBox(label), m_setting(setting), m_layer(target_layer), m_reverse(reverse)
+{
+  if (m_layer == nullptr)
```

Comment:
This is not a blocker to sign off but there is a lot of duplication in the code.  It's a pain to remove (I've been tinkering with ideas for a while).  Admittedly most of this was from before your change but there is even more now.  I wonder if we could come up with some mechanism to remove the duplication.  One idea is to pattern things similarly to `ToolTipWidget.h`.  Ex:

```
template <class Derived>
class ConfigControl : public Derived
{
public:
  void ConnectConfig()
  {
    connect(&Settings::Instance(), &Settings::ConfigChanged, this, [this] {
      QFont bf = font();
      bf.setBold(IsConfigBold());
      setFont(bf);

      const QSignalBlocker blocker(this);
      ConfigChangedUpdate();
    });
  }

protected:
  virtual bool IsConfigBold() const;
  virtual void ConfigChangedUpdate();
};
```

Which for example might look like:

```
class ConfigBool final : public ConfigControl<ToolTipCheckBox>
{
public:
  ConfigBool(...)
  {
    if (m_layer == nullptr)
    {
      setChecked(Config::Get(m_setting) ^ reverse);
      ConnectConfig();
     }
     else
     {
        setChecked(m_layer->Get(m_setting) ^ reverse)
        ConnectConfig();
     }
  }

  bool IsConfigBold() const override
  {
    if (m_layer == nullptr)
      return Config::GetActiveLayerForConfig(m_setting) != Config::LayerType::Base;
    return m_layer->Exists(m_setting.GetLocation());
  }

  void ConfigChangedUpdate() override
  {
    if (m_layer == nullptr)
      setChecked(Config::Get(m_setting) ^ m_reverse);
    else
      setChecked(m_layer->Get(m_setting) ^ m_reverse);
  }
};
```

If you passed in the layer there, you could possibly shift over more.  For instance `mousePressEvent` is duplicated in every class, you might be able to move that.

Is it worth it?  Not sure.  Just thought I'd throw out this idea.

Reply:
Yeah I was pondering this. You also presented it better than I would. It's probably worth while. I could also have Config::Get() take a layer = nullptr, so it could be used for both.

Reply:
It doesn't know what to do with setFont() and font().  Not sure how to solve that.